conda/shell/etc/profile.d/conda.csh

```diff
@@ -4,43 +4,49 @@
 # or in ~/.cshrc add the line
 #   source <CONDA_ROOT>/etc/profile.d/conda.csh
 
-# This block should only be for dev work. Under normal installs, _CONDA_EXE will be templated
-# in at the top of this file.
+# This block should only be for dev work. Under normal installs, _CONDA_EXE
+# will be templated in at the top of this file.
 if (! $?_CONDA_EXE) then
   set _CONDA_EXE="${PWD}/conda/shell/bin/conda"
 else
-  if ("$_CONDA_EXE" == "") then
-      set _CONDA_EXE="${PWD}/conda/shell/bin/conda"
-  endif
+    if ("$_CONDA_EXE" == "") then
+        set _CONDA_EXE="${PWD}/conda/shell/bin/conda"
+    endif
 endif
 
 if (`alias conda` == "") then
     if ($?_CONDA_ROOT) then
-        alias conda source $_CONDA_ROOT/etc/profile.d/conda.csh
+        alias conda source "${_CONDA_ROOT}/etc/profile.d/conda.csh"
     else
-        alias conda source $PWD/conda/shell/etc/profile.d/conda.csh
+        alias conda source "${PWD}/conda/shell/etc/profile.d/conda.csh"
     endif
     setenv CONDA_SHLVL 0
+
+    # setting to empty such that things don't break weirdly in
+    # non-interactive cases where (T)CSH doesn't set the prompt variable
     if (! $?prompt) then
         set prompt=""
     endif
 else
     switch ( $1 )
         case "activate":
-            eval `$_CONDA_EXE shell.csh activate "$2" $argv[3-]`
+            set ask_conda="`(setenv prompt '${prompt}' ; '${_CONDA_EXE}' shell.csh activate '${2}' ${argv[3-]})`" || exit ${status}
+            eval "${ask_conda}"
             rehash
             breaksw
         case "deactivate":
-            eval `$_CONDA_EXE shell.csh deactivate "$2" $argv[3-]`
+            set ask_conda="`(setenv prompt '${prompt}' ; '${_CONDA_EXE}' shell.csh deactivate '${2}' ${argv[3-]})`" || exit ${status}
+            eval "${ask_conda}"
             rehash
             breaksw
         case "install" | "update" | "uninstall" | "remove":
-            $_CONDA_EXE $argv[1-]
-            eval `$_CONDA_EXE shell.csh reactivate`
+            ${_CONDA_EXE:q} ${argv[1-]:q}
+            set ask_conda="`(setenv prompt '${prompt}' ; '${_CONDA_EXE}' shell.csh reactivate)`" || exit ${status}
+            eval "${ask_conda}"
             rehash
             breaksw
         default:
-            $_CONDA_EXE $argv[1-]
+            ${_CONDA_EXE:q} ${argv[1-]:q}
```

Comment:
I don't know `csh`, so can't really comment on those changes. I guess it makes sense to put that in a separate PR, so that it's easier for a C-shell-knowing person to review.

Reply:
CSH's `:q` variable modifier properly preserves quoting within the variables. I was trying to do this same trick for the `'${2}' ${argv[3-]}` but due to the sub-shelling it doesn't work (CSH is fun...).

As for another PR to improve review, from my experience with Conda (and just looking at how long CSH issues remain unaddressed in Conda) it will not be easy to find another CSH user.